scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/authentication/ManageUsers.java

```diff
@@ -0,0 +1,579 @@
+
+        opt = new Option(DELETE_OPTION, DELETE_OPTION_NAME, true,
+                "Action to delete an existing user");
+        opt.setArgName(DELETE_OPTION_NAME.toUpperCase());
+        opt.setArgs(0);
+        opt.setRequired(false);
+        optionGroup.addOption(opt);
+        options.addOptionGroup(optionGroup);
+
+        opt = new Option(LOGIN_OPTION, LOGIN_OPTION_NAME, true,
+                "Generate credentials for this specific user, will be asked interactively if not specified");
+        opt.setArgName(LOGIN_OPTION_NAME.toUpperCase());
+        opt.setArgs(1);
+        opt.setRequired(false);
+        options.addOption(opt);
+
+        opt = new Option(PWD_OPTION, PWD_OPTION_NAME, true,
+                "Password of the user, if the user is created or updated, will be asked interactively if not specified");
+        opt.setArgName(PWD_OPTION_NAME.toUpperCase());
+        opt.setArgs(1);
+        opt.setRequired(false);
+        options.addOption(opt);
+
+        opt = new Option(GROUPS_OPTION, GROUPS_OPTION_NAME, true,
+                "A comma-separated list of groups the user must be member of. Can be used when the user is created or updated");
+        opt.setArgName(GROUPS_OPTION_NAME.toUpperCase());
+        opt.setArgs(1);
+        opt.setRequired(false);
+        options.addOption(opt);
+
+        optionGroup.setRequired(false);
+        opt = new Option(KEYFILE_OPTION, KEYFILE_OPTION_NAME, true, "Public key path on the local filesystem [default:" +
+                getPublicKeyFilePath() + "]");
+        opt.setArgName(KEYFILE_OPTION_NAME.toUpperCase());
+        opt.setArgs(1);
+        opt.setRequired(false);
+        options.addOption(opt);
+
+        opt = new Option(LOGINFILE_OPTION, LOGINFILE_OPTION_NAME, true,
+                "Path to the login file in use [default:" + loginFilePath + "]");
+        opt.setArgName(LOGINFILE_OPTION_NAME.toUpperCase());
+        opt.setArgs(1);
+        opt.setRequired(false);
+        options.addOption(opt);
+
+        opt = new Option(GROUPFILE_OPTION, GROUPFILE_OPTION_NAME, true,
+                "Path to the group file in use [default:" + groupFilePath + "]");
+        opt.setArgName(GROUPFILE_OPTION_NAME.toUpperCase());
+        opt.setArgs(1);
+        opt.setRequired(false);
+        options.addOption(opt);
+
+
+        CommandLineParser parser = new DefaultParser();
+        CommandLine cmd = null;
+        try {
+            cmd = parser.parse(options, args);
+        } catch (Exception e) {
+            exitWithErrorMessage(newline + e.getMessage() + newline, "type -h or --help to display help screen", null);
+        }
+        return cmd;
+    }
+
+    private static void updateUserGroups(String login, String[] groups, Multimap<String, String> groupsMap) {
+        groupsMap.removeAll(login);
+        for (String group : groups) {
+            groupsMap.put(login, group);
+            System.out.println("Added group " + group + " to user " + login);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Replace this usage of System.out or System.err by a logger. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS106)